diff/difference_engine.py

```diff
@@ -140,18 +165,20 @@ def flips(prev_build, next_build):
 
             ['test1', 'testY']
     """
-    tests_prev = rm_params_from_names(prev_build['tests'])
-    tests_next = rm_params_from_names(next_build['tests'])
-    validate_build_contents(tests_prev, tests_next)
-    tests_prev_set = set(hashable(tests_prev))
-    tests_next_set = set(hashable(tests_next))
-    diff = tests_next_set - tests_prev_set
-    name_intersect = (set(util.get_names(tests_prev_set)) &
-                      set(util.get_names(tests_next_set)))
-    return [test[0] for test in diff if test[0] in name_intersect]
-
-
-#@timeit
+    tests_prev_passed = set(rm_params_from_names(prev_build['tests']['pass']))
+    tests_prev_failed = set(rm_params_from_names(prev_build['tests']['fail']))
+    tests_next_passed = set(rm_params_from_names(next_build['tests']['pass']))
+    tests_next_failed = set(rm_params_from_names(next_build['tests']['fail']))
+
+    diff = ((tests_next_passed - tests_prev_passed) |
+            (tests_next_failed - tests_prev_failed))
+
+    name_intersect = (set((list(tests_prev_passed) + list(tests_prev_failed))) &
```

Comment:
do you really think it's a good idea to do such a conversion? set(list(set()) looks kind of redundant to me.